dockstore-webservice/src/main/java/io/dockstore/webservice/core/Workflow.java

```diff
@@ -75,8 +75,8 @@
     @NamedQuery(name = "io.dockstore.webservice.core.Workflow.countAllPublished", query = "SELECT COUNT(c.id)" + Workflow.PUBLISHED_QUERY),
     @NamedQuery(name = "io.dockstore.webservice.core.Workflow.findByPath", query = "SELECT c FROM Workflow c WHERE c.sourceControl = :sourcecontrol AND c.organization = :organization AND c.repository = :repository"),
     @NamedQuery(name = "io.dockstore.webservice.core.Workflow.findByPathWithoutUser", query = "SELECT w FROM Workflow w WHERE w.sourceControl = :sourcecontrol AND w.organization = :organization AND w.repository = :repository and :user not in elements(w.users) "),
-    @NamedQuery(name = "io.dockstore.webservice.core.Workflow.findPublishedByOrganizationAndRepository", query = "SELECT w FROM Workflow w WHERE w.sourceControl = :sourcecontrol AND w.organization = :organization AND w.repository = :repository and w.isPublished = true"),
-    @NamedQuery(name = "io.dockstore.webservice.core.Workflow.findPublishedBySourceControl", query = "SELECT w FROM Workflow w WHERE w.sourceControl = :sourcecontrol and w.isPublished = true"),
+    @NamedQuery(name = "io.dockstore.webservice.core.Workflow.findPublishedByOrganizationAndRepository", query = "SELECT w FROM Workflow w WHERE w.sourceControl = :sourcecontrol AND w.organization = :organization AND w.repository = :repository and w.isPublished = true and (:allPublished = true or w.latestReleaseDate > :releaseDate)"),
```

Comment:
I don't understand what `allPublished` is doing, may need better name or documentation